src/Directory.BeforeCommon.targets

```diff
@@ -122,9 +122,10 @@
     <DefineConstants>$(DefineConstants);FEATURE_RUNTIMEINFORMATION</DefineConstants>
     <DefineConstants>$(DefineConstants);USE_MSBUILD_DLL_EXTN</DefineConstants>
     <DefineConstants>$(DefineConstants);WORKAROUND_COREFX_19110</DefineConstants>
+    <DefineConstants Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">$(DefineConstants);NET6_PLUS</DefineConstants>
```

Comment:
IIRC the SDK defines something much like this. Can we use that instead?

Reply:
running `rg -Fi "<defineconstants" -tmsbuild` in the sdk doesn't look promising. Sounds like something we can merge into the SDK to allow compiler directives in code for breaking changes between sdk versions. Something like:
```xml
<DefineConstants Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netcoreapp3.1'))">$(DefineConstants);NET31_PLUS</DefineConstants>
<DefineConstants Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0'))">$(DefineConstants);NET5_PLUS</DefineConstants>
<DefineConstants Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">$(DefineConstants);NET6_PLUS</DefineConstants>
```
@dsplaisted does this sound reasonable?

Questions answered with some greppin'
1. Where is DefineConstants even consumed?
   - Looks like it's passed as an argument to the compiler. https://github.com/dotnet/msbuild/blob/main/src/Tasks/Microsoft.CSharp.CurrentVersion.targets#L249

Reply:
https://github.com/dotnet/sdk/pull/14798 added it; in this case it's `NET6_0_OR_GREATER`.